sdk/ai/azure-ai-assistants/tests/model_assistants_test_base.py

```diff
@@ -0,0 +1,493 @@
+    #         credential=credential,
+    #         credential_scopes=credential_scopes,
+    #         headers=headers,
+    #         api_version=api_version,
+    #         logging_enable=LOGGING_ENABLED,
+    #     )
+
+    def _create_async_aoai_chat_client(
+        self, *, key_auth: bool = True, bad_key: bool = False, **kwargs
+    ) -> async_sdk.AssistantsClient:
+        endpoint, credential, credential_scopes, headers, api_version = self._load_aoai_chat_credentials(
+            key_auth=True, bad_key=bad_key, **kwargs
+        )
+        return async_sdk.AssistantsClient(
+            endpoint=endpoint,
+            credential=credential,
+            credential_scopes=credential_scopes,
+            headers=headers,
+            api_version=api_version,
+            logging_enable=LOGGING_ENABLED,
+        )
+
+    # def _create_async_embeddings_client(self, *, bad_key: bool = False, **kwargs) -> async_sdk.EmbeddingsClient:
+    #     endpoint, credential = self._load_embeddings_credentials(bad_key=bad_key, **kwargs)
+    #     return async_sdk.EmbeddingsClient(
+    #         endpoint=endpoint, credential=credential, logging_enable=LOGGING_ENABLED, **kwargs
+    #     )
+
+    # def _create_embeddings_client(self, *, bad_key: bool = False, **kwargs) -> sdk.EmbeddingsClient:
+    #     endpoint, credential = self._load_embeddings_credentials(bad_key=bad_key, **kwargs)
+    #     return sdk.EmbeddingsClient(endpoint=endpoint, credential=credential, logging_enable=LOGGING_ENABLED, **kwargs)
+
+    # def _create_embeddings_client_with_chat_completions_credentials(self, **kwargs) -> sdk.EmbeddingsClient:
+    #     endpoint = kwargs.pop("azure_ai_chat_endpoint")
+    #     key = kwargs.pop("azure_ai_chat_key")
+    #     credential = AzureKeyCredential(key)
+    #     return sdk.EmbeddingsClient(endpoint=endpoint, credential=credential, logging_enable=LOGGING_ENABLED)
+
+    def request_callback(self, pipeline_request) -> None:
+        self.pipeline_request = pipeline_request
+
+    def _validate_embeddings_json_request_payload(self) -> None:
+        headers = self.pipeline_request.http_request.headers
+        print(f"Actual HTTP request headers: {self.pipeline_request.http_request.headers}")
+        print(f"Actual JSON request payload: {self.pipeline_request.http_request.data}")
+        assert headers["Content-Type"] == "application/json"
+        assert headers["Content-Length"] == "311"
+        assert headers["extra-parameters"] == "pass-through"
+        assert headers["Accept"] == "application/json"
+        assert headers["some_header"] == "some_header_value"
+        assert "MyAppId azsdk-python-ai-inference/" in headers["User-Agent"]
+        assert " Python/" in headers["User-Agent"]
+        assert headers["Authorization"] == "Bearer key-value"
+        assert self.pipeline_request.http_request.data == self.EMBEDDINGDS_JSON_REQUEST_PAYLOAD
+
+    def _validate_chat_completions_json_request_payload(self) -> None:

```

Comment:
We can also remove all chat completions references from this document as that chat completions concept is Azure AI Inference specific.

Of course, there could be some similarities in tests between these SDKs and I understand you want probably to see what ideas to reuse from the existing SDK tests. It is good idea.